Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -64,16 +126,103 @@ class PreplannedMapModel: ObservableObject, Identifiable {
         }
     }
     
-    /// A Boolean value indicating if download can be called.
-    var canDownload: Bool {
-        switch status {
-        case .notLoaded, .loading, .loadFailure, .packaging, .packageFailure,
-                .downloading, .downloaded:
-            false
-        case .packaged, .downloadFailure:
-            true
+    /// Updates the status based on the download result of the mobile map package.
+    func updateDownloadStatus(for downloadResult: Optional<Result<MobileMapPackage, any Error>>) {
+        switch downloadResult {
+        case .success(let mobileMapPackage):
+            status = .downloaded
+            self.mobileMapPackage = mobileMapPackage
+        case .failure(let error):
+            status = .downloadFailure(error)
+        case .none:
+            return
+        }
+    }
+    
+    /// Sets the mobile map package if downloaded locally.
+    func setMobileMapPackage() {
```

Comment:
this function should be private and just called as part of this type's own init. There is no reason that I can see that the `MapViewModel` should be calling it.